server/s3_global_bucket_index_metadata.cc

```diff
@@ -188,32 +187,6 @@ void S3GlobalBucketIndexMetadata::save_successful() {
 
   state = S3GlobalBucketIndexMetadataState::saved;
 
-  // attempt to save the KV in replica global bucket list index
-  if (!motr_kv_writer) {
-    motr_kv_writer =
-        motr_kvs_writer_factory->create_motr_kvs_writer(request, s3_motr_api);
-  }
-  motr_kv_writer->put_keyval(
-      replica_global_bucket_list_index_oid, bucket_name, this->to_json(),
-      std::bind(&S3GlobalBucketIndexMetadata::save_replica, this),
-      std::bind(&S3GlobalBucketIndexMetadata::save_replica, this));
-
-  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
-}
-
-void S3GlobalBucketIndexMetadata::save_replica() {
-  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
-
-  // PUT Operation pass even if we failed to put KV in replica index.
-  if (motr_kv_writer->get_state() != S3MotrKVSWriterOpState::created) {
-    s3_log(S3_LOG_ERROR, request_id, "Failed to save KV in replica index.\n");
-
-    s3_iem_syslog(LOG_INFO, S3_IEM_METADATA_CORRUPTED,
-                  "Failed to save KV in replica index for bucket: %s",
-                  bucket_name.c_str());
-  }
-  this->handler_on_success();
```

Comment:
On_Success handlers should somehow be called - pls fix the flow

Reply:
fixed